lib/pbench/server/database/database.py

```diff
@@ -21,24 +21,18 @@ def get_engine_uri(config, logger):
             )
             return None
 
-        # return f"postgresql+{psql_driver}://{psql_username}:{psql_password}@{psql_host}:{psql_port}/{psql_db}"
-
-    @staticmethod
-    def init_engine(server_config, logger):
-        try:
-            return create_engine(Database.get_engine_uri(server_config, logger))
-        except Exception:
-            logger.exception("Exception while creating a sqlalchemy engine")
-            return None
-
     @staticmethod
     def init_db(server_config, logger):
+        # Attach the logger to the base class for models to find
+        if not hasattr(Database.Base, "logger"):
+            Database.Base.logger = logger
+
         # Make sure all the models are imported before this function gets called
         # so that they will be registered properly on the metadata. Otherwise
         # metadata will not have any tables and create_all functionality will do nothing
 
         Database.Base.query = Database.db_session.query_property()
-
+        Database.db_session.remove()  # Remove the placeholder static session
```

Comment:
Not sure why we never saw this before, but I was getting a really confusing `/home/classic/dev/sqlalchemy/lib/sqlalchemy/orm/scoping.py:106: SAWarning: At least one scoped session is already present.  configure() can not affect sessions that have already been created.` message in the unit tests, and after a bunch of searching I found out that we shouldn't be trying to re-configure the static `Database.db_session` (at line 38), but rather replacing it by removing the static placeholder first. Maybe there's a better way; but this seems to work, and it avoids the annoying warning.

Reply:
I got nothing here, but I'll just say that it seems really odd that almost the first thing we do is to delete the session.  Should we be trying to avoid creating it during class instantiation, so that we don't have to delete it before (re)creating it as part of this instance?

Given that we're going to remove it at line 35, why are we fetching stuff from it at line 34?  Should that perhaps be using a different path to the value?

Reply:
I actually hadn't noticed that the preceding line was referencing the session. I have no idea whether that makes a difference... although I deployed on my VM and I was able to register and GET a user (which is a query), plus login and logout, and it all worked.

To be safe, I'll reverse the order here so we're fetching a query from the new scoped session created (I presume) by the `configure` below. This is pretty much, um, alchemy, to me. Too many moving pieces to track them all.